common/app/com/commercetools/sunrise/common/sessions/CacheableSessionCookieStrategy.java

```diff
@@ -0,0 +1,108 @@
+package com.commercetools.sunrise.common.sessions;
+
+import com.commercetools.sunrise.common.contexts.RequestScoped;
+import play.Configuration;
+import play.cache.CacheApi;
+import play.mvc.Http;
+
+import javax.inject.Inject;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Uses a session cookie to store string information about the user.
+ * Objects are first saved into the cache and uniquely associated with the user's session cookie with a generated session ID.
+ * For more information: <a href="https://www.playframework.com/documentation/2.5.x/JavaSessionFlash">Session in Play Framework</a>
+ */
+@RequestScoped
+public class CacheableSessionCookieStrategy extends SessionCookieStrategy implements TypedSessionStrategy {
+
+    private static final String DEFAULT_SESSION_ID_KEY = "sunrise-session-id";
+    private final String sessionIdKey;
+    private final CacheApi cacheApi;
+
+    @Inject
+    public CacheableSessionCookieStrategy(final Http.Session session, final CacheApi cacheApi, final Configuration configuration) {
+        super(session);
+        this.cacheApi = cacheApi;
+        this.sessionIdKey = configuration.getString("session.id", DEFAULT_SESSION_ID_KEY);
```

Comment:
Good catch, I gave it a quick name to proof the concept and then I forgot about it!